spec/rails_helper.rb

```diff
@@ -77,6 +77,14 @@
   config.include SidekiqTestHelpers
   config.include ElasticsearchHelpers
 
+  config.after(:each, type: :system) do
```

Comment:
This is 🔥 

The only thing that worries me a bit is that `_on_request` is considered private (the doc has a `:api: private` annotation), so it may change at any time. I wonder if [`Warden.test_mode!`](https://www.rubydoc.info/github/hassox/warden/Warden#test_mode!-class_method) in combination with [`Warden.test_reset!`](https://www.rubydoc.info/github/hassox/warden/Warden/Test/WardenHelpers#test_reset!-instance_method) would also solve our problem, I looked through some of my old Rails projects and that's the approach used there.